batch/batch/worker/worker.py

```diff
@@ -841,7 +841,12 @@ async def delete_container(self):
                 log.info(f'{self} container is still running, killing crun process')
                 self.process.terminate()
                 self.process = None
-                await check_exec_output('crun', 'kill', '--all', self.container_name, 'SIGTERM')
+                try:
+                    await check_exec_output('crun', 'state', self.container_name)
+                except CalledProcessError:
+                    log.info(f'container {self} cannot be killed because it does not exist', exc_info=True)
+                else:
+                    await check_exec_output('crun', 'kill', '--all', self.container_name, 'SIGTERM')
```

Comment:
We are *still* not guaranteed that the container will exist at this point, but it is probably worth checking to see how often that arises in practice.